airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -532,11 +532,12 @@ def find_pod(self, namespace: str, context: Context, *, exclude_checked: bool =
         ).items
 
         pod = None
-        num_pods = len(pod_list)
+        running_pods = [pod for pod in pod_list if pod.status.phase == "Running"]
```

Comment:
How about the pod which is in pending state? It will result in launching another application pod which is not need.

Reply:
Initial commit , I have set `find_pod` block after `await_pod_start` so we'll have only running pods but removed it later as recently introduced callbacks uses `remote_pod` for `on_pod_creation` in `execute_sync`. Now , I have added support for pending pods.